clustering/ejb/infinispan/src/main/java/org/wildfly/clustering/ejb/infinispan/Scheduler.java

```diff
@@ -35,6 +35,13 @@
      */
     void schedule(I id);
 
+    /**
+     * Invalidates a previously scheduled task for specified bean whose task
+     * will be rescheduled again.
+     * @param id a bean identifier
+     */
+    void prepareRescheduling(I id);
```

Comment:
Why create a new method?  Aren't we just changing the behavior of the existing cancel(...) method?

Reply:
We are not, because cancel will invoke ExpirationTracker.forget, that will remove all entries from ExpirationTracker. If we keep calling cancel like before the gain is relatively small. I'm trying to make sure we schedule or cancel only once per call, I'm not sure if that's entirely feasible, but I'm trying to at least decrease the amount of times we add items to the tracker, to prevent the cost of invoking currentTimeInMillis

Reply:
Ah, I see.  I assumed that there was no longer a use case for ExpirationTracker.forget, but there is.  So, cancel(I) is still triggered from cancel(Locaility, Locality) and from SchedulableBean.close() in the event that the bean is invalid (i.e. was removed).